packages/vuetify/src/components/VTextField/VTextField.tsx

```diff
@@ -0,0 +1,89 @@
+// Styles
+import './VTextField.sass'
+
+// Components
+import { VField } from '@/components/VField/VField'
+
+// Composables
+import { useIntersectionObserver } from '@/composables/intersectionObserver'
+import { useProxiedModel } from '@/composables/proxiedModel'
+
+// Utilities
+import { defineComponent, isComponentInstance } from '@/util'
+import { computed, ref } from 'vue'
+
+// Types
+import type { VFieldSlot } from '@/components/VField/VField'
+
+const dirtyTypes = ['color', 'file', 'time', 'date', 'datetime-local', 'week', 'month']
+
+export const VTextField = defineComponent({
+  name: 'VTextField',
+
+  inheritAttrs: false,
+
+  props: {
+    autofocus: Boolean,
+    prefix: String,
+    persistentPlaceholder: Boolean,
+    suffix: String,
+    type: {
+      type: String,
+      default: 'text',
+    },
+    modelValue: String,
+  },
+
+  emits: {
+    'update:modelValue': (val: string) => true as any,
+  },
+
+  setup (props, { attrs, slots }) {
+    const value = useProxiedModel(props, 'modelValue')
+
+    const internalDirty = ref(false)
+    const isDirty = computed(() => {
+      return internalDirty.value || !!value.value || dirtyTypes.includes(props.type)
+    })
+
+    const { intersectionRef, isIntersecting } = useIntersectionObserver((_, observer) => {
+      if (props.autofocus && isIntersecting) {
```

Comment:
1. `isIntersecting` is a ref so should be accessed with `.value`
2. the `useIntersectionObserver` API is maybe a bit sloppy here. isIntersecting is only extracted from the return to be accessed inside the callback where it could also be available. perhaps callback signature should be `(isIntersecting: boolean, entries: ..., observer: ...) => void`